### ISSUE TITLE
GridFS Deployment

### ISSUE THREAD

**Issue by appjitsu** [CollectionFS/Meteor-CollectionFS#902]
I have an app hosted on DigitalOcean using Compose.io and GridFS. ImageMagick is installed on the server.

Everything works and uploads great in development.

Then I deploy the app and copy my local collections over to the Compose.io database. I can view all uploaded images I copied over from local dev on the server. The problem is with uploads on the server.

The upload doesn't give an error either on the server or client, but the url is is undefined on any new file uploads. It does create new collection entries for the uploads.

Any help would be greatly appreciated.

```javascript
Template.usersEdit.helpers({
  photo: function() {
    let photo = Data.Collections.UserPhotos.findOne({
      userId: FlowRouter.getParam('_id')
    });
    console.log("photo: ", photo);
    console.log("url: ", photo.url("userPhotoRegThumbs"));
    return photo;
  },
});
```

The above outputs: 

```json
  FS.File {createdByTransform: true, _id: "ZhheBrGidBXhj8EWg", original: Object, userId: "PefChSfXxqbHgeCCp", uploadedAt: Wed Jan 06 2016 06:46:27 GMT-0600 (CST)…}
_id: "ZhheBrGidBXhj8EWg"
collection: FS.Collection
collectionName: "userPhotos"
copies: Object
createdByTransform: true
original: Object
uploadedAt: Wed Jan 06 2016 06:46:27 GMT-0600 (CST)
userId: "PefChSfXxqbHgeCCp"
__proto__: EventEmitter
  url: null
```

**Comment by appjitsu** [comment]
I do see this when I try to upload:

```javascript
The provided value 'undefined' is not a valid enum value of type XMLHttpRequestResponseType.
```

**Comment by appjitsu** [comment]
mongo version is 3.0.6 on compose.io. No meteor client side or server side errors. No mongo errors.

**Comment by appjitsu** [comment]
I upgraded the databse on the server to 3.0.6, but it didn't make a difference.

**Comment by appjitsu** [comment]
I cleared the compose.io db and uploaded a file. Still no url.

```json
{ 
    "_id" : "39YQh6YEaGKiyxXYs", 
    "original" : {
        "name" : "12366312_10153811398547417_3549822771242018005_n.jpg", 
        "updatedAt" : ISODate("2015-12-22T01:01:24.000+0000"), 
        "size" : NumberInt(108304), 
        "type" : "image/jpeg"
    }, 
    "userId" : "KnrGJAqd9Qhhunhoq", 
    "uploadedAt" : ISODate("2016-01-06T20:18:14.406+0000"), 
    "copies" : {
        "userPhotoOriginals" : {
            "name" : "12366312_10153811398547417_3549822771242018005_n.jpg", 
            "type" : "image/jpeg", 
            "size" : NumberInt(108304), 
            "key" : "568d76866ea8dc1b0006c150", 
            "updatedAt" : ISODate("2016-01-06T20:18:17.001+0000"), 
            "createdAt" : ISODate("2016-01-06T20:18:17.001+0000")
        }
    }
}
```

**Comment by nooitaf** [comment]
You could try this on server side and see if something pops up
```javascript
FS.debug = true;
```

**Comment by appjitsu** [comment]
Ok i'll give that a try.

**Comment by appjitsu** [comment]
TLDR; use https://github.com/numtel/meteor-cfs-image-resize/ if you are using mupx.

@nooitaf I turned on debugging and it showed that CollectionFS couldn't find GraphicsMagick or ImageMagick. I am using mupx for deployment and it uses docker. So I tried setting up mupx to install GrahphicsMagick inside the docket container, but failed to do so. Then I looked around for alternatives and found meteor-cfs-image-resize and it did the trick perfectly.

Thanks for your help!
